server/shared/src/main/scala/org/http4s/server/staticcontent/FileService.scala

```diff
@@ -33,39 +33,206 @@ import org.log4s.getLogger
 import org.typelevel.ci._
 
 import java.io.File
+import scala.annotation.nowarn
 import scala.util.control.NoStackTrace
 
 object FileService {
   private[this] val logger = getLogger
 
+  @deprecated("use FileService.Fs2PathCollector", "0.23.8")
   type PathCollector[F[_]] = (File, Config[F], Request[F]) => OptionT[F, Response[F]]
 
+  type Fs2PathCollector[F[_]] = (Path, Config[F], Request[F]) => OptionT[F, Response[F]]
+
   /** [[org.http4s.server.staticcontent.FileService]] configuration
     *
     * @param systemPath path prefix to the folder from which content will be served
     * @param pathPrefix prefix of Uri from which content will be served
-    * @param pathCollector function that performs the work of collecting the file or rendering the directory into a response.
+    * @param fs2PathCollector function that performs the work of collecting the file or rendering the directory into a response.
     * @param bufferSize buffer size to use for internal read buffers
     * @param cacheStrategy strategy to use for caching purposes. Default to no caching.
```

Comment:
```suggestion
    * @param fs2PathCollector function that performs the work of collecting the file or rendering the directory into a response.
    * @param pathPrefix prefix of Uri from which content will be served
    * @param cacheStrategy strategy to use for caching purposes. Default to no caching.
    * @param bufferSize buffer size to use for internal read buffers
```

Reply:
github won't allow me to apply this, will commit manually

Reply:
weird